app/models/concerns/profile_tailoring.rb

```diff
@@ -33,4 +33,19 @@ def update_os_minor_version(version)
 
     update!(os_minor_version: version)
   end
+
+  def tailoring_valid?
+    rule_groups_for_rules = RuleGroup.where(id: rules.includes(:rule_group)
+                                     .references(:rule_group).flat_map(&:rule_group_ancestor_ids))
+
+    conflicts = relationships_for(rules, rule_groups_for_rules, 'conflicts').map(&:right)
+    requires = relationships_for(rules, rule_groups_for_rules, 'requires').map(&:right)
+
+    valid_relationships?(rules, rule_groups_for_rules, requires, conflicts)
+  end
+
+  def valid_relationships?(rules, rule_groups, requires, conflicts)
+    rules_and_groups = rules | rule_groups
+    (requires - rules_and_groups).empty? && (conflicts - rules_and_groups) == conflicts
```

Comment:
@skateman I adjusted this method a little so that it still works correctly even if there are duplicates in `requires` or `conflicts` so we don't need to worry about using `distinct` anymore